### ISSUE TITLE
Instructions for how to use `.xcassets` colors

### ISSUE THREAD

**Issue by danielsaidi** [SwiftGen/SwiftGen#472]
I browsed through your old, closed issues and found that you have merged in a feature that lets you generate Swift code for a color asset catalog, but there are no instructions on the home page for how you use this feature. I would love a sample script for this. When I use the same syntax as for strings or images, I just get an error that says that my `Colors.xcassets` folder is not a valid path.

**Comment by AliSoftware** [comment]
Hi,

Color assets are used exactly the same way as any other asset (especially images) parsed from `.xcassets` catalogs.
SwiftGen parse every type of asset from `*.xcassets` catalogs you ask to parse, and produce a generated file containing constants for all assets found in each of those assets catalogs, images, colors (and, in next release, data assets as well)

So the example isn't really different from the usage with images. Even if I agree that we could add an example usage in the README for colors too, there are such examples in the documentation specific to each template. For example if you're using the `swift4` template for xcassets (`swiftgen xcassets -t swift4 path/to/your/Colors.xcassets`, or similar setup using a config file) you can find the associated documentation for that `swift4` `xcassets` template [here](https://github.com/SwiftGen/SwiftGen/blob/master/Documentation/templates/xcassets/swift4.md) alongside example usages on how to use those generated colors for constants in your Swift code.

**Comment by danielsaidi** [comment]
Hi,

Yeah, I have tried the command you specify above with both the latest SwiftGen version and the legacy version we use in our old project (with L10n enum), but both says that the color asset catalog isn't a valid path. I'll try again tomorrow and close this issue if I make it work.

Thanks!

**Comment by AliSoftware** [comment]
Huh yeah strange indeed then.
Which version of SwiftGen are you using?

I guess you triple checked already the spelling of the `Colors.xcassets` file name and that it is a valid asset catalog (like, not an empty folder or whatnot but with an actual Contents.json file and subfolders in it and all)?

Can't think of another idea for now 🧐🤔

**Comment by AliSoftware** [comment]
Maybe share your `swiftgen.yml` config file and an extract of your file hierarchy to give us more clues? 🤔

**Comment by danielsaidi** [comment]
Good morning,

Thank you for your suggestions. I have looked at this more this morning, and think I've found the reason for this behavior.

I run `5.3.0` in my personal projects. At work, however, we still use `3.0.0` since we still use the old `L10n` enums. We will refactor this some day :)

If I copy the `images` build step in the work project:

```
$PODS_ROOT/SwiftGen/bin/swiftgen images -t swift3 --output ${PROJECT_DIR}/BookBeat/Assets/Images.swift ${PROJECT_DIR}/BookBeat/Assets/Images.xcassets
``` 

and change it to look like this:

```
$PODS_ROOT/SwiftGen/bin/swiftgen colors -t swift3 --output ${PROJECT_DIR}/BookBeat/Assets/Colors.swift ${PROJECT_DIR}/BookBeat/Assets/Colors.xcassets
```

I get the following build error:

```
`/Users/admin/Projects/bookbeat/ios-app/BookBeat/Assets/Colors.xcassets` is not a `file`
```

However, this is not true. The asset folder exists and the path is valid.

If I move to the terminal and adjust the script to look like this (still running 3.0.0):

```
Pods/SwiftGen/bin/swiftgen colors -t swift3 --output BookBeat/Assets/Colors.swift BookBeat/Assets/Colors.xcassets
```

I get the same error. If I adjust the script to use the system install of SwiftGen (5.0):

```
swiftgen colors -t swift3 --output BookBeat/Assets/Colors.swift BookBeat/Assets/Colors.xcassets
```

I instead get the following error:

```
error: Unsupported file type for BookBeat/Assets/Colors.xcassets. The supported file types are: txt, clr, json, xml
```

So, I guess the `is not a file` error is just a side effect of using the really old version. 

However, I now noticed that you used `xcassets` for colors as well. I completely overlooked this.

Case (and issue) closed :)

**Comment by AliSoftware** [comment]
Thanks for the update.

Yes back in 3.0 we didn't support colors in xcassets yet. But now we do

1) `swiftgen colors` is the old command (still present in 5.3+ though) which expected a txt, xml or clr file. We still kept it because some people might still prefer to use that input format for colors, especially the xml input format which is the one used by Android so that allows people to share colors between iOS and Android projects.

2) At the time of 3.0 we only supported images in xcassets but not yet colors, and the command for parsing images in xcassets catalogs was named `swiftgen images`

3) But since then (4.0 or something?) we started to support other types in xcassets, not only images but also colors (and will soon also support data assets too) so we renamed the command `swiftgen xcassets` now that it parses more stuff from those xcassets than just images.

---

Side note: if you missed the feature change before, maybe you also missed that we now support a config file as well instead of passing all those parameters via the command line.
Be sure to check the CHANGELOG to follow all the announcements 👍
